cmd/sovereignnode/incomingHeader/incomingSCRs.go

```diff
@@ -0,0 +1,82 @@
+package incomingHeader
+
+import (
+	"encoding/hex"
+	"math/big"
+
+	"github.com/multiversx/mx-chain-core-go/core"
+	"github.com/multiversx/mx-chain-core-go/data"
+	"github.com/multiversx/mx-chain-core-go/data/smartContractResult"
+	"github.com/multiversx/mx-chain-go/process"
+)
+
+const (
+	minTopicsInEvent  = 4
+	numTransferTopics = 3
+)
+
+type scrInfo struct {
+	scr  *smartContractResult.SmartContractResult
+	hash []byte
+}
+
+func (ihp *incomingHeaderProcessor) createIncomingSCRs(events []data.EventHandler) ([]*scrInfo, error) {
+	scrs := make([]*scrInfo, 0, len(events))
+
+	for _, event := range events {
+		topics := event.GetTopics()
+		if len(topics) < minTopicsInEvent || len(topics[1:])%numTransferTopics != 0 {
+			log.Error("incomingHeaderHandler.createIncomingSCRs",
+				"error", errInvalidNumTopicsIncomingEvent,
+				"num topics", len(topics),
+				"topics", topics)
+			continue
+		}
+
+		scr := &smartContractResult.SmartContractResult{
+			RcvAddr: topics[0],
```

Comment:
this is a little different than on the mainnet, where sender and receiver are the same in case of a ESDTNFTTransfer. Do you still need topics[0] in the scr.Data?

Reply:
Yes, it is different as per specifications defined in the PR description:
> senderAddress -> ESDTSystemSCAddress
> recvAddress -> address

Should be fine to be different from mainnet, since we will have different:
- scr pre processor
- scr processor
- MultiESDTNFTTransfer builtin function

Reply:
ok